libs/common/src/database/model/database.model.ts

```diff
@@ -0,0 +1,44 @@
+import { Transform, Type } from 'class-transformer';
+import { IsInt, IsNotEmpty, IsOptional, IsString, Min } from 'class-validator';
+import { FindOptionsWhere, ObjectID } from 'typeorm';
+import { transformSortType } from './transform.model';
+
+export class FindOptions {
+    @IsInt()
+    @Min(1)
+    @Type(() => Number)
+    page: number;
+
+    @IsInt()
+    @Min(1)
+    @Type(() => Number)
+    max: number;
+
+    @IsOptional()
+    @IsString()
+    @IsNotEmpty()
+    search?: string;
+
+    @IsOptional()
+    @Transform(({ value }) => {
+        return transformSortType(value);
+    })
```

Comment:
Eu sugiro você criar um decorator que faça a validação pra você:

```suggestion
    @IsSortParams()
```